### ISSUE TITLE
"result_is_hit( execute_state -> result )"  with multiple targets

### ISSUE THREAD

**Issue by navv1234** [simulationcraft/simc#1610]
Originally reported on Google Code with ID 1611

```
TLDR: doesnt work at all, needs fixing.

Long Version: 
Take a look at warrior_t::mortal_strike. It is a typical ability that triggers effects
if it hits the target: In this case also the deep wounds dot.

Now take a look at warrior_t::thunder_clap_t: It also procs deep wounds, but because
it is an aoe ability, it should apply it to every target, but the code doesn't. Instead
on execute() it enters the default execute() routines, applies the damage of the ability
to every target, then forgets all but the last execute_state and returns to thunder_clap_t::execute().
It then checks for a "hit" and applies the deep wounds effect to its primary target.

This problem should occur for any "spread effect to others" such as some of the rogue
dots, the fire mage fire_blast, etc.

I could think of several (bad) ways to change the behavior, but as this has some bigger
implications, I leave it to the veterans to discuss the best.

```

Reported by `dr.max.moellers` on 2013-03-18 14:16:29
